pkg/ebpf/btf.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/cilium/ebpf/btf"
+	"github.com/mholt/archiver/v3"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/host"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func GetBTF(userProvidedBtfPath, collectionPath string) (*btf.Spec, CoReResult) {
+	var btfSpec *btf.Spec
+	var err error
+
+	if userProvidedBtfPath != "" {
+		btfSpec, err = loadBTFFrom(userProvidedBtfPath)
+		if err == nil {
+			log.Debugf("loaded BTF from %s", userProvidedBtfPath)
+			return btfSpec, successCustomBTF
+		}
+	}
+
+	btfSpec, err = checkEmbeddedCollection(collectionPath)
+	if err == nil {
+		log.Debugf("loaded BTF from embedded collection")
+		return btfSpec, successEmbeddedBTF
+	}
+	log.Debugf("couldn't find BTF in embedded collection: %s", err)
+
+	btfSpec, err = btf.LoadKernelSpec()
+	if err == nil {
+		log.Debugf("loaded BTF from default kernel location")
+		return btfSpec, successDefaultBTF
+	}
+	log.Debugf("couldn't find BTF in default kernel locations: %s", err)
+
+	return nil, btfNotFound
+}
+
+func checkEmbeddedCollection(collectionPath string) (*btf.Spec, error) {
```

Comment:
This is assuming the entire BTF collection is in a folder. That is a lot of files. I think it would be better to have the collection be a single tarball. We can walk the tarball to extract the specific file we want.